miraclelinux/meta-debian-extended
# unbound: Update version from 1.9.0-2+deb10u2 to 1.9.0-2+deb10u3

# Purpose of pull request
This version fixes following CVEs.
- CVE-2020-28935
- CVE-2022-3204
- CVE-2022-30698
- CVE-2022-30699

DLA: DLA 3371-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check unbound

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " unbound"
```
```
$ bitbake core-image-minimal
```

### Run qemu image
```
$ runqemu nographic qemuarm64
```

### Check unbound
Add 'local-data' entry for host-PC in unbound.conf.
```
# echo -e "server:\n        local-data: \"host-pc    IN A <HOST-PC-IPADDR>\"" >> /etc/unbound/unbound.conf
# tail -n 3 /etc/unbound/unbound.conf
```
Checks unbound.conf file for errors.
```
# /usr/sbin/unbound-checkconf
```
Restart the unbound daemon.
```
# /etc/init.d/unbound restart
```
Query domain name of 'host-PC' by nslookup command.
```
# nslookup host-pc
```

## Test result
### Build package
Build succeeded.
```
$ bitbake unbound
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |##############################################################################################################################################| Time: 0:00:23
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2384 targets, 80 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:04469d4b27cb5203de34211f3fd8248a510b6c35"
meta-debian-extended = "update-unbound_1.9.0-2+deb10u3:8444766faa928ea197db1a897fd3799704d37efe"
meta-emlinux         = "HEAD:c0c24adb1bdb7f4e479976541314e9d9f1bb1f10"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |###########################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 24 Found 10 Missed 14 Current 413 (41% match, 96% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2010 tasks of which 1965 didn't need to be rerun and all succeeded.
```

### Package test
Name resolution for added "host-pc" can be resolved.
The unbound daemon is working.

In my qemu environment, IP address of 'host-pc' is 192.168.7.2.
```
root@qemuarm64:~# echo -e "server:\n        local-data: \"host-pc    IN A 192.168.7.2\"" >> /etc/unbound/unbound.conf
root@qemuarm64:~# tail -n 3 /etc/unbound/unbound.conf
#     redis-timeout: 100
server:
        local-data: "host-pc    IN A 192.168.7.2"
root@qemuarm64:~# /usr/sbin/unbound-checkconf
unbound-checkconf: no errors in /etc/unbound/unbound.conf
root@qemuarm64:~# /etc/init.d/unbound restart
Stopping unbound: 
Starting unbound: 
root@qemuarm64:~# nslookup host-pc
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost.localdomain

Name:      host-pc
Address 1: 192.168.7.2
```
